srcs/Client.cpp

```diff
@@ -34,14 +34,70 @@ const Client::State& Client::getState()
 	return (state);
 }
 
+const std::string&		Client::getUsername()
+{
+	return (username);
+}
+
+int						Client::setUsername(std::string name)
+{
+	if (state == LOCKED || state == UNINITIALIZED)
+	{
+		return (-1);//ERR_NOTREGISTERED
+	}
+	if (state == SET)
```

Comment:
Should we be able to change the username after this state?

Reply:
There is if conditions which checks before setting username. If password or nickname wasnt provided, then it returns -1 and doesnt set username